src/expr.cpp

```diff
@@ -4530,7 +4530,7 @@ static std::pair<llvm::Constant *, bool> lGetExprListConstant(const Type *type,
     bool isVaryingInit = false;
     bool isNotValidForMultiTargetGlobal = false;
     if (exprs.size() == 1 && (CastType<AtomicType>(type) != nullptr || CastType<EnumType>(type) != nullptr ||
-                              CastType<PointerType>(type) != nullptr)) {
+                              CastType<PointerType>(type) != nullptr) && exprs[0] != nullptr) {
         if (isStorageType) {
```

Comment:
I would check the corner case for `exprs[0]` inside `if` statement and return `std::pair<llvm::Constant *, bool>(nullptr, false);` if it is `nullptr`. Otherwise, we might start exploring some states within this function that weren't possible before the fix.